bumble/profiles/hap.py

```diff
@@ -272,24 +272,12 @@ def on_connection(connection: Connection) -> None:
             def on_disconnection(_reason) -> None:
                 self.currently_connected_clients.remove(connection)
 
-            # TODO Should we filter on device bonded && device is HAP ?
-            self.currently_connected_clients.add(connection)
-            if (
-                connection.peer_address
-                not in self.preset_changed_operations_history_per_device
-            ):
-                self.preset_changed_operations_history_per_device[
-                    connection.peer_address
-                ] = []
-                return
-
-            async def on_connection_async() -> None:
-                # Send all the PresetChangedOperation that occur when not connected
-                await self._preset_changed_operation(connection)
-                # Update the active preset index if needed
-                await self.notify_active_preset_for_connection(connection)
+            @connection.on('pairing')  # type: ignore
```

Comment:
One issue here is that this only remembers paired connections when the pairing happens while the service is running. But what about connections from devices that were previously paired and bonded?

Reply:
If you mean any pairing when the HAS is not running ?

I assumed that setting up HAS is done before we connect to a remote device.

If there was a device already paired, it will go within 
```python
            if connection.peer_resolvable_address:
                self.on_incoming_paired_connection(connection)
```

directly

Reply:
A problem here is that peer_resolvable_address may not be set if identity information is not exchanged (though Android almost always does that). I didn't find a clear identity requirement in HAP (besides IAC https://www.bluetooth.com/specifications/specs/hap-1-0/ ), so I am afraid checking RPA is not very precise here.

Since what HAP requires is an LTK, maybe we can make on_connection an async function (using `utils.AsyncRunner`) and check whether LTK is available via `device.get_long_term_key(...)`?

Reply:
Example:

```py
    def __init__(
        self, device: Device, features: HearingAidFeatures, presets: List[PresetRecord]
    ) -> None:
        ...
        device.on('connection', self.on_connection)

    @AsyncRunner.run_in_task()
    async def on_connection(self, connection: Connection) -> None:
        def on_disconnection(reason: int) -> None:
            del reason  # unused
            self.currently_connected_clients.remove(connection)

        connection.on('disconnection', on_disconnection)

        # TODO Should we filter on device bonded && device is HAP ?
        self.currently_connected_clients.add(connection)
        if (
            connection.peer_address
            not in self.preset_changed_operations_history_per_device
        ):
            self.preset_changed_operations_history_per_device[
                connection.peer_address
            ] = []
            return

        # Send all the PresetChangedOperation that occur when not connected
        await self._preset_changed_operation(connection)
        # Update the active preset index if needed
        await self.notify_active_preset_for_connection(connection)
```

Reply:
> But what about connections from devices that were previously paired and bonded?

In what scenario do we have previously paired and bonded devices without HAS knowing it ? I believe HAS is present since the device is "alive".

Could we open a bug to track this specific behavior instead and implement it later ? We would need this PR to land to merge some HAP Avatar test within Android.